ledger/participant-integration-api/src/main/scala/platform/store/cache/StateCache.scala

```diff
@@ -60,7 +69,15 @@ private[platform] case class StateCache[K, V](cache: Cache[K, V], registerUpdate
             }
             .getOrElse(Long.MinValue)
 
-        if (competingLatestForKey < validAt) cache.put(key, value) else ()
+        if (competingLatestForKey < validAt) putInternal(key, value, validAt) else ()
+        if (validAt >= _cacheIndex) {
+          _cacheIndex = validAt
+        } else {
+          logger.warn(
+            "Incoming synchronous update at an index before the cache index. Flushing the cache.."
```

Comment:
```suggestion
            "Unexpected incoming synchronous update at an index before the cache index. Flushing the cache.."
```

I'm expecting that we do not expect to see this warning do we?

Reply:
No, we're not. Good suggestion

Reply:
I'm not yet set on the proper action here. Invalidate only for key, invalidate all or kill the participant. Still thinking over it.

Reply:
Went with only invalidating the key and aborting the pending updates for the key.

Reply:
Dropped cache and pending updates invalidation. Please see later comments